src/routes/vendor.tsx

```diff
@@ -22,8 +22,15 @@ export default function VendorPage() {
   const [productFilter, setProductFilter] = useState('')
   const [locationFilter, setLocationFilter] = useState('')
   const { setShowLoadingOverlay } = useCommonStore()
+  const [sortBy, setSortBy] = useState<'name' | 'area_group_name' | 'rating' | ''>('');
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('asc');
+
   const { vendors, metadata, isSuccess, error } = useGetVendorsQuery(
-    { page, product: productFilter, location: locationFilter },
+    { page, 
+      product: productFilter, 
+      location: locationFilter, 
+      order_by: sortBy, 
+      order: sortOrder },
```

Comment:
> This PR also implemented the sorting logic right? I don't find where is the logic to sort the data apart from just altering the arrow UI. Can you show me the reference of the logic I'm looking for?

[![image](https://github.com/user-attachments/assets/d6b4a598-0254-4715-a356-c5c6291b0e42)](https://github.com/kg-procurement/kg-procurement/blob/main/internal/vendors/accessor.go#L112-L116)

The backend already includes the sorting logic as part of the API implementation. It supports sorting through pagination arguments such as `OrderBy` and `Order`. To enable the sorting functionality, we just need to pass these arguments as part of the API parameters.